lib/cucumber/rails/capybara/select_dates_and_times.rb

```diff
@@ -23,30 +24,43 @@ def select_date(date, options)
         # Select a Rails time. Options hash must include from: +label+
         def select_time(time, options)
           time = Time.zone.parse(time)
-          if ::Rails::VERSION::MAJOR >= 7
-            # Rails 7 generates date fields using input type="time". Capybara support's them
+          base_dom_id = get_base_dom_from_options(options)
+
+          # Rails 7 use HTML5 input type="time" by default. If input is not present fallback to plain select boxes alternative.
+          # It's safe to use has_css? without waiting/retry. We already know field's label is visible
+          if html5_input_field_present?(base_dom_id)
             fill_in options[:from], with: time
           else
-            base_dom_id = get_base_dom_id_from_label_tag(options[:from])
-
             find(:xpath, ".//select[@id='#{base_dom_id}_4i']").select(time.hour.to_s.rjust(2, '0'))
             find(:xpath, ".//select[@id='#{base_dom_id}_5i']").select(time.min.to_s.rjust(2,  '0'))
           end
         end
 
         # Select a Rails datetime. Options hash must include from: +label+
         def select_datetime(datetime, options)
-          if ::Rails::VERSION::MAJOR >= 7
-            # Rails 7 generates datetime fields using input type="datetime-local". Capybara support's them
+          base_dom_id = get_base_dom_id_from_label_tag(options[:from])
+
+          # Rails 7 use HTML5 input type="datetime-local" by default. If input is not present fallback to plain select boxes alternative.
+          # It's safe to use has_css? without waiting/retry. We already know field's label is visible
+          if html5_input_field_present?(base_dom_id)
             fill_in options[:from], with: DateTime.parse(datetime)
           else
-            select_date(datetime, options)
-            select_time(datetime, options)
+            extended_options = options.merge(base_dom_id: base_dom_id)
```

Comment:
Can you explain this bit here. Everything else is fine

Reply:
It's an optimization. I'm passing down `base_dom_id` to `select_date` and `select_time` so they don't need to find the label and extract the select/input id again.

Reply:
Ah right I've spotted it now. It's not really extending the options, that's where I got confused. It's added the existing dom id to the options hash and then adding a guard clause to the get_base_dom_from_options method.

Although I'm not sure what I'd call it though. I think this area needs a lot of TLC and fixes but for now this will do. Thanks for explanation